### DISCUSSION TITLE
How to call the Ocelot gateway from Angular?

### DISCUSSION THREAD

**Discussion by Ramakrish** [ThreeMammals/Ocelot#1999] [Q&A] [answered]
Hi Team 

We are implemented the Ocelot in dot net core project..

We are created the reroutes for two different kind of API's.

While running the  Ocelot with reroutes in my browsers it provide the expected result.

We need to call the same Ocelot API gateway reroute URL from angular 7 project.

### Is it possible to call Ocelot API Gateway URL from Angular 7?

Kindly help me on that..

**Comment by Loongle** [comment]
you need configuration your stratup. additional cors configuration

**Reply by raman-m** [reply]
Yes. Extra setup should be done.
And we still have problems with OPTIONS and CORS
- #799

**Comment by raman-m** [comment, marked as answer]
Hi @Ramakrish! 
Hare Krishna! Hare Rama! :smile: Forever!

> Is it possible to call Ocelot API gateway URL from angular 7??

Angular is front-end framework. Any front-end framework requires to deploy special type of BFF gateway which has special configuration and setup.
In general Ocelot is good choice to deploy BFF gateway instance: we support main protocols: `http` + `https`, `ws` + `wss`.
We have some problems with OPTIONS and CORS policies, see #799. But we believe, experienced ASP.NET developer can solve these problems.

**Reply by raman-m** [reply]
@ggnaegi  Could you share your Angular vs Ocelot experience too please? :wink:
